### PR TITLE
DEV-3637: Support Redshift data source

### PR DESCRIPTION
**Pull request by saiharshavellanki**
- Added support for Redshift data source
- Updated the documentation

Screenshot 2024-04-01 at 12 57 43 PM
Screenshot 2024-04-01 at 1 29 54 PM
Screenshot 2024-03-16 at 6 42 10 PM
Screenshot 2024-03-16 at 6 42 44 PM

### CODE REVIEW

File: docs/pages/api-reference/sources/redshift.md [hunk 1]
```diff
@@ -1,53 +1,38 @@
...
-<Expandable title="account" type="str">
-
-Snowflake account identifier. This is the first part of the URL used to access 
-Snowflake. For example, if the URL is `https://<account>.snowflakecomputing.com`, 
-then the account is `<account>`. 
-
-This is usually of the form `<ORG_ID>-<ACCOUNT_ID>`. Refer to the 
-[Snowflake documentation](https://docs.snowflake.com/en/user-guide/admin-account-identifier#finding-the-organization-and-account-name-for-an-account) 
-to find the account identifier.
+<Expandable title="s3_access_role_arn" type="str">
```
Comment (mohitreddy1996) [inline docs/pages/api-reference/sources/redshift.md:14]: I think we need to provide some more information around this. 

Potentially provide some IAM policy JSON, trust relationships etc to make our customer's experience smoother.

**Review by mohitreddy1996 — APPROVED** [review]
Please check with @nikhilgarg28 to get the documentation in. 

My recommendation is we should provide at least the JSON for trust relationship and IAM policy the IAM role which the redshift cluster can assume to write to s3. This will help our customers setup source and dataset easily

File: docs/pages/api-reference/sources/redshift.md (hunk 1 above)
Comment (saiharshavellanki) [reply docs/pages/api-reference/sources/redshift.md:14]: Got it. Will check with @nikhilgarg28 on this and update the documentation

Comment (saiharshavellanki) [reply docs/pages/api-reference/sources/redshift.md:14]: Updated the documentation with steps to add s3_access_role_arn. Added sample policy JSON as well

**Pull request merged**